tools/log-collector/log_collector.go

```diff
@@ -33,49 +35,24 @@ type LogCollector struct {
 	k8sClientSet *kubernetes.Clientset
 }
 
-// setLogsAndClient sets user bases log level for the log collector such as INFO, ERROR, DEBUG etc
-func (l *LogCollector) setLogsAndClient() error {
+// setClient initialize clients
+func (l *LogCollector) setClient() {
 	l.k8sClient, l.disClient, l.k8sClientSet = getClient()
 	l.disClient.LegacyPrefix = "/api/"
-
-	// Setting Log Level
-	level, lErr := log.ParseLevel(l.Loglevel)
-	if lErr != nil {
-		log.Errorf("Unable to Parse Log Level : %s", lErr.Error())
-		return lErr
-	}
-	log.SetLevel(level)
-
-	return nil
 }
 
 // CollectLogsAndDump collects call all the related resources of triliovault
 func (l *LogCollector) CollectLogsAndDump() error {
 
-	lErr := l.setLogsAndClient()
-	if lErr != nil {
-		return lErr
-	}
-	log.Info("Checking Namespaces")
-	coreGV, cgErr := l.getAPIGVResources(CoreGv)
-	if cgErr != nil {
-		return cgErr
-	}
-	namespaceResource := getResourceByName(coreGV, Namespaces)
-	namespaceObjects := l.getResourceObjects(getAPIGroupVersionResourcePath(CoreGv), &namespaceResource)
-	allNamespaces := getObjectsNames(namespaceObjects)
+	l.setClient()
+	l.getClusterNodes()
 
-	if len(l.Namespaces) != 0 && !l.isSubset(allNamespaces) {
-		log.Error("Specified namespaces doesn't exists in the cluster")
+	nsErr := l.checkNamespaces()
+	if nsErr != nil {
+		log.Errorf("%s", nsErr.Error())
 		return nil
```

Comment:
why are you returning `nil` here?

Also, log error of `CollectLogsAndDump` func.

Reply:
resolved